horovod/tensorflow/xla_mpi_ops.cc

```diff
@@ -570,6 +571,536 @@ XLA_REGISTER_CUSTOM_CALL_TARGET(CallbackHVDAllreduceDone, "CUDA");
 } // namespace tensorflow
 } // namespace horovod
 
-#endif // TENSORFLOW_VERSION >= 2006000000
 #endif // HAVE_CUDA
+#if HAVE_ROCM
```

Comment:
The following block is virtually identical to the one for `HAVE_CUDA`. Differences are limited to `cuda` -> `hip` string replacements, but they are spread over a few hundred lines. I think for now this repetition is OK, but it will be a good idea to reduce the duplication of code in the future.